assembl/static2/js/app/components/login/signupForm.jsx

```diff
@@ -150,30 +152,65 @@ class SignupForm extends React.Component<SignupFormProps, SignupFormState> {
               })}
             {hasTermsAndConditions && (
               <FormGroup className="left margin-left-2">
-                <Checkbox checked={this.state.checked} type="checkbox" onChange={this.toggleCheck} required inline>
+                <Checkbox
+                  checked={termsAndConditionsIsChecked}
+                  type="checkbox"
+                  onChange={() => this.toggleCheck('termsAndConditions')}
+                  required
+                  inline
+                >
                   <Translate value="termsAndConditions.iAccept" />
                   <a
                     onClick={(e) => {
                       e.preventDefault();
-                      const Terms = (
-                        <TermsForm handleAcceptButton={this.handleAcceptButton} isChecked={this.state.checked} lang={lang} />
+                      const content = (
+                        <LegalForm
+                          handleAcceptButton={this.handleAcceptButton}
+                          checked={termsAndConditionsIsChecked}
+                          text={termsAndConditionsText}
+                          legalContentsType="termsAndConditions"
+                        />
                       );
-                      displayCustomModal(Terms, true, 'modal-large');
+                      displayCustomModal(content, true, 'modal-large');
                     }}
                   >
                     <Translate value="termsAndConditions.link" className="terms-link" />
                   </a>
                 </Checkbox>
               </FormGroup>
             )}
+            {hasPrivacyPolicy && (
+              <FormGroup className="left margin-left-2">
+                <Checkbox
+                  checked={privacyPolicyIsChecked}
+                  type="checkbox"
+                  onChange={() => this.toggleCheck('privacyPolicy')}
+                  required
+                  inline
+                >
+                  <Translate value="privacyPolicy.iAccept" />
+                  <a
+                    onClick={(e) => {
+                      e.preventDefault();
+                      const content = (
+                        <LegalForm
+                          text={privacyPolicyText}
+                          handleAcceptButton={this.handleAcceptButton}
+                          checked={privacyPolicyIsChecked}
+                          legalContentsType="privacyPolicy"
+                        />
+                      );
+                      displayCustomModal(content, true, 'modal-large');
```

Comment:
I would have put these in methods to avoid confusion. Someone who reads the code to quickly may think that `<LegalForm>` is rendered directly in the render method (same for the others). I guess that we could also create an intermediate component that renders the `FormGroup` + `Checkbox` since the code is very similar for tac and privacy policy.